python_modules/dagster/dagster/_core/definitions/time_window_partitions.py

```diff
@@ -29,45 +30,80 @@ class TimeWindow(NamedTuple):
 class TimeWindowPartitionsDefinition(
     PartitionsDefinition[TimeWindow],  # pylint: disable=unsubscriptable-object
     NamedTuple(
-        "_TimeWindowPartitions",
+        "_TimeWindowPartitionsDefinition",
         [
-            ("schedule_type", PublicAttr[ScheduleType]),
             ("start", PublicAttr[datetime]),
             ("timezone", PublicAttr[str]),
             ("fmt", PublicAttr[str]),
             ("end_offset", PublicAttr[int]),
-            ("minute_offset", PublicAttr[int]),
-            ("hour_offset", PublicAttr[int]),
-            ("day_offset", PublicAttr[Optional[int]]),
+            ("cron_schedule", PublicAttr[str]),
         ],
     ),
 ):
+    """
+    A set of partitions where each partitions corresponds to a time window.
+
+    The provided cron_schedule determines the bounds of the time windows. E.g. a cron_schedule of
+    "0 0 * * *" will result in daily partitions that start at midnight and end at midnight of the
+    following day.
```

Comment:
the time window here is not necc. uniform here right? Like if you do 1AM and 3AM every Tuesday, you get some 2-hour long windows and some week-minus-two-hours long windows.

Would that be an argument for not including both start and end time and just including start time? Or is it still better to include both for consistency?

Reply:
When you say "including", where do you mean? Do you mean like the general notion that this corresponds to a time window instead of a moment in time? Or arguments that are accepted somewhere?